builtin/aws/ecs/platform.go

```diff
@@ -1133,64 +1133,64 @@ func (p *Platform) Destroy(
 		return err
 	}
 
-	var listener *elbv2.Listener
-
+	// search all listeners for the deployments TargetGroupArn
 	if len(listeners.Listeners) > 0 {
-		listener = listeners.Listeners[0]
-
-		log.Debug("listener arn", "arn", *listener.ListenerArn)
-
-		def := listener.DefaultActions
+		for _, listener := range listeners.Listeners {
+			log.Debug("listener arn", "arn", *listener.ListenerArn)
 
-		var tgs []*elbv2.TargetGroupTuple
+			def := listener.DefaultActions
 
-		// If there is only 1 target group, delete the listener
-		if len(def) == 1 {
-			log.Debug("only 1 target group, deleting listener")
-			_, err = elbsrv.DeleteListener(&elbv2.DeleteListenerInput{
-				ListenerArn: listener.ListenerArn,
-			})
+			var tgs []*elbv2.TargetGroupTuple
 
-			if err != nil {
-				return err
-			}
-		} else if len(def) > 1 && def[0].ForwardConfig != nil {
-			// Multiple target groups means we can keep the listener
-			var active bool
+			if def[0].ForwardConfig != nil {
```

Comment:
Better to check if it is nil, and if so, continue the loop. That will reduce the indentation depth too.

Reply:
This line will always panic if def is empty, so that needs to be checked for.

Reply:
Updated with nil checks, and cleaned up the indentation depth as well.